testing/fil_token_integration/actors/basic_token_actor/src/lib.rs

```diff
@@ -214,14 +231,12 @@ pub fn invoke(params: u32) -> u32 {
                 // FRCXXX Token standard
                 3839021839 => {
                     // Mint
-                    // This is an exmaple mint function which simply gives the caller 100 tokens
-                    let minter = caller_address();
-                    token_actor
-                        .util
-                        .mint(&minter, &minter, &TokenAmount::from(100), &Default::default())
-                        .unwrap();
-                    token_actor.util.flush().unwrap();
-                    NO_DATA_BLOCK_ID
+                    let params = deserialize_params(params);
+                    let res = token_actor.mint(params).unwrap();
+
+                    let cid = token_actor.util.flush().unwrap();
+                    sdk::sself::set_root(&cid).unwrap();
```

Comment:
Without this, other calls to the actor will continue to use the previous state. I think most of the other methods that change state need to do this too?

Reply:
I hadn't considered this. It seems that it would be necessary to set the new root in order to handle re-entrant calls. 

However, doesn't this need to happen before the receiver hook is called (i.e. within `mint`) rather than after `mint` since the execution flow is passed to the called actor by the VM?

Reply:
Yes, we'll need to set root before the receiver hook. Fun!

Alex, maybe we need to split up the mint/transfer behaviour into two parts, or take a callback that persists state, or something like that? The library can't do it internally.

Reply:
Mint and Transfer currently both flush state before calling the receiver hook but don't set the root at that time. Splitting mint/transfer and the receiver hook calls might be the nicest way to do things so any state persistence stuff can be handled outside of that (so token mutates state, but never stores/loads).

I've been having thoughts about making load, flush/store, and maybe a transaction thing like this into a trait for state structs, it should be generic enough that we could write the one implementation and just `impl PersistentState for MyState {}`?
The transaction option would save state (and set the root) and then revert if necessary based on the result of a callback, so we'd call the receiver hook there. I guess there might be other uses for that beyond receiver hooks too?

minting would then look something like this:
```Rust
let mint = state.token.mint(...)?;
state.call_or_revert(call_receiver_hook(mint));
```

Reply:
> I've been having thoughts about making load, flush/store, and maybe a transaction thing like this into a trait

Put that onto a list for possible developer tooling things to do later. It's beyond scope for the tokens, but I'm keen to gather ideas like this.